src/services/gcs/backend.rs

```diff
@@ -0,0 +1,744 @@
+            let mut m = ObjectMetadata::default();
+            m.set_mode(ObjectMode::DIR);
+
+            debug!("backed root object stat finished");
+            return Ok(m);
+        }
+
+        let resp = self.get_object_metadata(&p).await?;
+
+        if resp.status().is_success() {
+            let mut m = ObjectMetadata::default();
+
+            if let Some(v) = parse_content_length(resp.headers())
+                .map_err(|e| other(ObjectError::new("stat", &p, e)))?
+            {
+                m.set_content_length(v);
+            }
+
+            if let Some(v) =
+                parse_etag(resp.headers()).map_err(|e| other(ObjectError::new("stat", &p, e)))?
+            {
+                m.set_etag(v);
+                m.set_content_md5(v.trim_matches('"'));
+            }
+
+            if let Some(v) = parse_last_modified(resp.headers())
+                .map_err(|e| other(ObjectError::new("stat", &p, e)))?
+            {
+                m.set_last_modified(v);
+            }
+
+            if p.ends_with('/') {
+                m.set_mode(ObjectMode::DIR);
+            } else {
+                m.set_mode(ObjectMode::FILE);
+            };
+
+            debug!("object {} stat finished: {:?}", &p, m);
+            Ok(m)
+        } else if resp.status() == StatusCode::NOT_FOUND && p.ends_with('/') {
+            let mut m = ObjectMetadata::default();
+            m.set_mode(ObjectMode::DIR);
+
+            debug!("object {} stat finished", &p);
+            Ok(m)
+        } else {
+            let err =
+                parse_error_response("stat", args.path(), parse_error_status_code, resp).await;
+            warn!("object {} stat: {:?}", args.path(), err);
+            Err(err)
+        }
+    }
+
+    #[trace("delete")]
+    async fn delete(&self, args: &OpDelete) -> Result<()> {
+        increment_counter!("opendal_gcs_delete_requests");
+
+        let p = self.get_abs_path(args.path());
+        debug!("object {} delete start", &p);
+
+        let resp = self.delete_object(&p).await?;
+
+        if resp.status().is_success() {
+            debug!("object {} delete finished", &p);
+            Ok(())
+        } else {
+            let err =
+                parse_error_response("delete", args.path(), parse_error_status_code, resp).await;
+            warn!("object {} delete: {:?}", args.path(), err);
+            Err(err)
+        }
+    }
+
+    #[trace("list")]
+    async fn list(&self, args: &OpList) -> Result<DirStreamer> {
+        increment_counter!("opendal_gcs_list_requests");
+
+        let mut path = self.get_abs_path(args.path());
+        // Make sure list path is endswith '/'
+        if !path.ends_with('/') && !path.is_empty() {
+            path.push('/')
+        }
+        debug!("object {} list start", &path);
+
+        Ok(Box::new(DirStream::new(Arc::new(self.clone()), &path)))
+    }
+
+    fn presign(&self, args: &OpPresign) -> Result<PresignedRequest> {
```

Comment:
Please don't implement this function as reqsign doesn't support query sign for gcs.

And this function will leak user's oauth2 token.